tgui/packages/tgui/interfaces/FishAnalyzer.tsx

```diff
@@ -0,0 +1,332 @@
+                                                  />
+                                                </Stack.Item>
+                                              </Stack>
+                                            </Flex.Item>
+                                            <Flex.Item>
+                                              <DmIcon
+                                                mt={1}
+                                                icon={evolution.evolution_icon}
+                                                icon_state={
+                                                  evolution.evolution_icon_state
+                                                }
+                                                height="32px"
+                                                width="32px"
+                                              />
+                                            </Flex.Item>
+                                          </Flex>
+                                        </Stack.Item>
+                                      ),
+                                    )}
+                                  </Stack>
+                                )}
+                              </Stack.Item>
+                            </Stack>
+                          </Flex.Item>
+                        </Flex>
+                      </Stack.Item>
+                      <Stack.Item>
+                        <Flex>
+                          <Flex.Item
+                            width="25%"
+                            grow
+                            style={{
+                              color: fish.fish_suitable_temp
+                                ? '#1ac400'
+                                : 'red',
+                            }}
+                          >
+                            {fish.fish_min_temp}k - {fish.fish_max_temp}k
+                          </Flex.Item>
+                          <Flex.Item
+                            ml={3}
+                            grow
+                            style={{
+                              color: fish.fish_fluid_compatible
+                                ? '#1ac400'
+                                : 'red',
+                            }}
+                          >
+                            {DisectName(fish.fish_fluid_type)}
+                          </Flex.Item>
+                        </Flex>
+                      </Stack.Item>
+                    </Stack>
+                  </Stack.Item>
+                ))}
+              </Stack>
+            </Section>
+          </Stack.Item>
+        </Stack>
+      </Window.Content>
+    </Window>
+  );
+};
+
+function DisectName(input: string): string {
+  if (input.length > 12) {
+    const firstSpace = input.indexOf(' ');
+
+    if (!firstSpace) {
+      return input.substring(0, 12);
+    }
+    return input.substring(0, firstSpace);
+  }
+  return input;
+}
```

Comment:
could just simplify this to
```suggestion
function dissectName(input: string): string {
  return input.split(' ').slice(0, 12);
}
```
also, functions are camelCase. only types and components get pascal case like this